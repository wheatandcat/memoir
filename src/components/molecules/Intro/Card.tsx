import React, { memo } from 'react';
import {
  StyleSheet,
  ImageBackground,
  ImageSourcePropType,
  TouchableWithoutFeedback,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Image from 'components/atoms/Image';
import theme from 'config/theme';

export type Props = {
  source: ImageSourcePropType;
  text: string;
  onNext: () => void;
};

const Card: React.FC<Props> = (props) => {
  return (
    <ImageBackground
      source={require('../../../img/common/frame.png')}
      resizeMode="cover"
      style={styles.image}
    >
      <View style={styles.inner}>
        <View>
          <Image
            source={props.source}
            width={95}
            height={95}
            resizeMode="contain"
          />
        </View>
        <View mt={4}>
          <Text
            textAlign="center"
            variants="large"
            lineHeight={50}
            fontFamily="NotoSansJP-Bold"
          >
            {props.text}
          </Text>
        </View>
        <View mt={4}>
          <TouchableWithoutFeedback onPress={props.onNext}>
            <View style={styles.nextButton}>
              <Text fontWeight="bold" textAlign="center">
                次へ
              </Text>
            </View>
          </TouchableWithoutFeedback>
        </View>
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  image: {
    flex: 1,
    justifyContent: 'center',
    position: 'relative',
  },
  inner: {
    alignItems: 'center',
    height: '50%',
  },
  nextButton: {
    width: 180,
    backgroundColor: theme().color.background.main,
    paddingVertical: theme().space(3),
  },
});

export default memo(Card);
