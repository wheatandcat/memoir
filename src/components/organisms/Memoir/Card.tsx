import React, { memo } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  useWindowDimensions,
  ViewStyle,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Category from 'components/atoms/Category';
import theme from 'config/theme';
import Image from 'components/atoms/Image';
import IconButton from 'components/molecules/IconButton';
import { Item } from 'queries/api/index';

type User = {
  id: string;
  name: string;
};

export type Props = {
  title: Item['title'];
  categoryID: Item['categoryID'];
  user: User;
  onPress: () => void;
};

const Card: React.FC<Props> = (props) => {
  const windowWidth = useWindowDimensions().width;

  const titleStyle: ViewStyle = {
    width: windowWidth - 130,
  };

  return (
    <TouchableOpacity onPress={props.onPress}>
      <View style={styles.root}>
        <View>
          <Category categoryID={props.categoryID} />
        </View>
        <View pl={3}>
          <View style={[styles.title, titleStyle]}>
            <Text numberOfLines={2} ellipsizeMode="tail" lineHeight={25}>
              {props.title}
            </Text>
          </View>
          <View style={styles.user}>
            <Image
              source={require('../../../img/icon/account.png')}
              width={20}
              height={20}
            />
            <View pl={2}>
              <Text variants="small">{props.user?.name}</Text>
            </View>
          </View>
        </View>
        <View>
          <Text textAlign="center" size="sm">
            0
          </Text>
          <IconButton
            name="favorite"
            size="base"
            color={theme().color.base.main}
            onPress={() => null}
          />
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: theme().space(2),
  },
  title: {},
  user: {
    paddingVertical: theme().space(2),
    flexDirection: 'row',
    alignItems: 'center',
  },
});
