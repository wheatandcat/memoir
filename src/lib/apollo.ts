import { setContext } from '@apollo/client/link/context';
import { Alert } from 'react-native';
import { onError } from '@apollo/client/link/error';
import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
import { storageKey, getItem } from 'lib/storage';
import Auth from 'lib/auth';

export const cache = new InMemoryCache();
const auth = new Auth();

type Param = {
  Authorization?: string;
  UserID?: string;
};

const makeApolloClient = async () => {
  const uri = `${process.env.API_HOST}/query`;

  const authLink = setContext(async (_, { headers }) => {
    const h = headers;
    const param: Param = {};
    const token = await auth.getIdToken();

    if (token) {
      param.Authorization = `Bearer ${token}`;
    } else {
      const userID = await getItem(storageKey.USER_ID_KEY);

      if (userID) {
        param.UserID = userID;
      }
    }

    return {
      headers: {
        ...h,
        ...param,
      },
    };
  });

  const errorLink = onError((error) => {
    console.log(error.graphQLErrors);
    if ((error.graphQLErrors || []).length > 0) {
      const graphQLErrors = error.graphQLErrors || [
        { message: 'エラー発生しました' },
      ];

      const message = graphQLErrors[0].message;
      Alert.alert('エラー', message);
    }
  });

  return new ApolloClient({
    link: errorLink.concat(authLink.concat(createHttpLink({ uri }))),
    cache: cache,
  });
};

export default makeApolloClient;
