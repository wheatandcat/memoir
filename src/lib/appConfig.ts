import firebase from 'lib/system/firebase';

export type AppConfig = {
  maintenance: boolean;
  maintenanceMessage: string;
  maintenancePeriod: string | null;
  supportVersion: string;
};

export const defaultAppConfig = () => ({
  maintenance: false,
  maintenanceMessage: '',
  maintenancePeriod: null,
  supportVersion: '1.0.0',
});

export const getAppConfig = async (
  db: firebase.firestore.Firestore
): Promise<AppConfig> => {
  const qs = await db.collection('appConfig').limit(1).get();

  const records = qs.docs.map((elem) => {
    return elem.data();
  });

  if (!records || !records[0]) {
    return defaultAppConfig();
  }

  return records[0] as AppConfig;
};
