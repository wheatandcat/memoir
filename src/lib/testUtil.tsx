import React from 'react';
import { GraphQLHandler, GraphQLRequest } from 'msw';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { render } from '@testing-library/react-native';
import { server } from 'mocks/server';
import fetch from 'cross-fetch';

const link = createHttpLink({
  uri: 'http://localhost:8080/query',
  fetch,
  credentials: 'same-origin',
});

const client = new ApolloClient({
  link,
  uri: 'http://localhost:8080/query',
  cache: new InMemoryCache(),
});

export const testRenderer =
  (children: React.ReactNode) =>
  (responseOverride?: GraphQLHandler<GraphQLRequest<never>>) => {
    if (responseOverride) {
      server.use(responseOverride);
    }
    return render(<ApolloProvider client={client}>{children}</ApolloProvider>);
  };
