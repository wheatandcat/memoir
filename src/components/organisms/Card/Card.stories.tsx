import React from 'react';
import { mockFn } from 'storyBookUtils/index';
import Card from './Card';

export default {
  title: 'organisms/Card',
};

export const _Card = () => (
  <Card title="本を読んだ" categoryID={1} onPress={mockFn('onPress')} />
);
