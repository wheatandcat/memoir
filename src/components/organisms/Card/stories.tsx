import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { mockFn } from 'storyBookUtils/index';
import Card from './Card';

storiesOf('organisms/Card', module).add('Card', () => (
  <Card onPress={mockFn('onPress')} />
));
