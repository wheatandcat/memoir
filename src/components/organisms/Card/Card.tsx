import React, { memo } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  useWindowDimensions,
  ViewStyle,
} from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Category from 'components/atoms/Category';
import theme from 'config/theme';
import Image from 'components/atoms/Image';
import { Item } from 'queries/api/index';
import setting from 'components/atoms/Category/setting';
import { categoryBorderStyle } from 'lib/category';

type User = {
  id: string;
  name: string;
};

export type Props = {
  title: Item['title'];
  categoryID: Item['categoryID'];
  user?: User;
  onPress: () => void;
};

const Card: React.FC<Props> = (props) => {
  const windowWidth = useWindowDimensions().width;

  const titleStyle: ViewStyle = {
    width: windowWidth - 120,
  };
  if (props.user) {
    titleStyle.height = 60;
    titleStyle.justifyContent = 'flex-end';
  }

  const rootStyle = { height: 75 };
  if (props.user) {
    rootStyle.height = 95;
  }

  const category = setting().icon.find(
    (v) => v.id === props.categoryID
  )?.category;
  const categoryStyle: ViewStyle[] = [
    rootStyle,
    styles.root,
    categoryBorderStyle(category || 0),
  ];

  return (
    <TouchableOpacity onPress={props.onPress}>
      <View style={categoryStyle}>
        <View mx={2}>
          <Category categoryID={props.categoryID} />
        </View>
        <View>
          <View style={[styles.title, titleStyle]}>
            <Text numberOfLines={2} ellipsizeMode="tail" lineHeight={25}>
              {props.title}
            </Text>
          </View>
          {!!props.user?.id && (
            <View style={styles.user}>
              <Image
                source={require('../../../img/icon/account.png')}
                width={20}
                height={20}
              />
              <View pl={2}>
                <Text variants="small">{props.user?.name}</Text>
              </View>
            </View>
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.light,
    flexDirection: 'row',
    alignItems: 'center',
    borderLeftWidth: 5,
  },
  title: {
    paddingLeft: theme().space(0),
  },
  user: {
    paddingVertical: theme().space(2),
    flexDirection: 'row',
    alignItems: 'center',
  },
});
