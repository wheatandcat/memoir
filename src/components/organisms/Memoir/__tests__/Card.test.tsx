import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import Card, { Props } from '../Card';

const propsData = (): Props => ({
  title: 'title',
  categoryID: 1,
  user: {
    id: 'test',
    displayName: 'suzuki',
    image: '',
  },
});

describe('components/organisms/Memoir/Card.tsx', () => {
  it('正常にrenderすること', () => {
    testRenderer(<Card {...propsData()} />)();
    expect(screen.findAllByText('suzuki')).toBeTruthy();
  });
});
