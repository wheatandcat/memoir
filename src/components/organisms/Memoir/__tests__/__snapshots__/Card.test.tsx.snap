// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/Memoir/Card.tsx 正常にrenderすること 1`] = `
<ForwardRef
  onPress={[MockFunction]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 8,
      }
    }
  >
    <View>
      <Memo(Category)
        categoryID={1}
      />
    </View>
    <View
      pl={3}
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "width": 620,
            },
          ]
        }
      >
        <Text
          ellipsizeMode="tail"
          fontFamily="RobotoCondensed-Bold"
          lineHeight={25}
          numberOfLines={2}
          variants="body"
        >
          title
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingVertical": 8,
          }
        }
      >
        <Image
          height={20}
          source={null}
          width={20}
        />
        <View
          pl={2}
        >
          <Text
            fontFamily="RobotoCondensed-Bold"
            variants="small"
          >
            test
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Text
        fontFamily="RobotoCondensed-Bold"
        size="sm"
        textAlign="center"
        variants="body"
      >
        0
      </Text>
      <Memo(IconButton)
        color="#D8D7D6"
        name="favorite"
        onPress={[Function]}
        size="base"
      />
    </View>
  </View>
</ForwardRef>
`;
